### PR TITLE
Reduce minimum Python version to 3.8

### LINKED ISSUE #26: Reduce python requirements to 3.8
As per https://github.com/informalsystems/modelator-py/pull/65, apparently I'll need to pull the minimum python version back down to support integration with MBT code bases. This will require some amount of fiddling with code to drop the newer idioms.

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #26 

Follows https://github.com/informalsystems/modelator/pull/283, and required to
ensure MBT can use this library.

Ignore the lock file in review: it's a actually a very tiny diff.

### CODE REVIEW

File: chai/cmd_executor.py [hunk 1]
```diff
@@ -140,12 +140,6 @@ def _checking_err(data: dict) -> CmdExecutorCheckError:
         return _typechecking_err(data)
 
 
-def _config_json(source: Source, cfg: Optional[dict]) -> str:
-    # TODO: error if source already set in config?
-    config = cfg or {}
-    return json.dumps(config | source.to_dict())
```
Comment (shonfeder) [inline chai/cmd_executor.py:146]: This was a dead function.

**Comment by shonfeder** [comment]
I've also sneaked a couple unrelated fixes in, since this small PR seems like a good place to address them.

**Comment by shonfeder** [comment]
The failing integration test is still just #24. I've figured out a fix for it, which I'll add in a follow-up.

**Review by thpani — COMMENTED** [review]
LGTM :)

**Pull request merged**